v2/blockstore/ro_blockstore.go

```diff
@@ -0,0 +1,190 @@
+package blockstore
+
+import (
+	"bufio"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"golang.org/x/exp/mmap"
+	"io"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	carv1 "github.com/ipld/go-car"
+	"github.com/ipld/go-car/util"
+	"github.com/ipld/go-car/v2/internal/index"
+	internalio "github.com/ipld/go-car/v2/internal/io"
+)
+
+var _ blockstore.Blockstore = (*ReadOnlyBlockStore)(nil)
+
+// errUnsupported is returned for unsupported operations
+var errUnsupported = errors.New("unsupported operation")
+
+type (
+	// ReadOnlyBlockStore provides a read-only Car Block Store.
+	ReadOnlyBlockStore struct {
+		backing io.ReaderAt
+		idx     index.Index
```

Comment:
we probably want a non-internal Index type - it sounds like the sharded blockstore would like to be able to detach indexes to store them separate from cars, and also be able to recombine a carv1 + index into a carv2

Reply:
https://github.com/ipld/go-car/issues/89